pkg/osquery/extension.go

```diff
@@ -320,9 +320,23 @@ func (e *Extension) Enroll(ctx context.Context) (string, bool, error) {
 		return "", true, errors.Wrap(err, "generating UUID")
 	}
 
-	enrollDetails, err := getEnrollDetails(e.osqueryClient)
-	if err != nil {
-		return "", true, errors.Wrap(err, "query enrollment details")
+	// We've seen this fail, so add some retry logic. Keep trying to
+	// get enroll details, until either we succeed, or we;ve tried 5
+	// times.
+	var enrollDetails service.EnrollmentDetails
+EnrollDetailsLoop:
```

Comment:
is this the `gofmt` indentation level? I've not seen this syntax yet, so I don't know..

EDIT: Yep